remill/Arch/AArch64/Arch.cpp

```diff
@@ -719,6 +719,28 @@ static bool DecodeBitMasks(uint64_t N /* one bit */,
   }
   return true;
 }
+/// Utility function for extracting [From, To] bits from a uint32_t.
+static inline unsigned slice(uint32_t Bits, unsigned From, unsigned To) {
+  assert(From < 32 && To < 32 && From >= To);
+  return (Bits >> To) & ((1 << (From - To + 1)) - 1);
+}
+
+// Adapted from LLVM 
+// https://llvm.org/svn/llvm-project/llvm/branches/ggreif/CallInst-operands/lib/Target/ARM/Disassembler/ARMDisassemblerCore.cpp
+static uint64_t VFPExpandImm(uint8_t imm, RegClass float_class) {
+  assert(float_class == kRegS || float_class == kRegD || float_class == kRegH);
+  uint64_t result;
+
+  unsigned bit6 = slice(imm, 6, 6);
+  if(kRegD == float_class) {
+    result = (uint64_t)slice(imm, 7, 7) << 63 | (uint64_t)slice(imm, 5, 0) << 48;
```

Comment:
Coding style forbids C-style casts. Use `static_cast` in this case. But realistically, you can rewrite `slice` to operand and return `uint64_t`s, thereby avoiding casting.